src/aio.rs

```diff
@@ -390,10 +460,27 @@ async fn connect_simple<T: Connect>(
     Ok(match *connection_info.addr {
         ConnectionAddr::Tcp(ref host, port) => {
             let socket_addr = get_socket_addrs(host, port)?;
-
             <T>::connect_tcp(socket_addr).await?
         }
 
+        #[cfg(feature = "tls")]
+        ConnectionAddr::TcpTls {
+            ref host,
+            port,
+            insecure,
+        } => {
+            let socket_addr = get_socket_addrs(host, port)?;
+            <T>::connect_tcp_tls(host, socket_addr, insecure).await?
+        }
+
+        #[cfg(not(feature = "tls"))]
+        ConnectionAddr::TcpTls { .. } => {
```

Comment:
We shouldn't even have this variant if it's not compiled with `tls`, right?

Reply:
well, it's like the `unix` one, it's mostly because we want to parse the address string and return an error if the user didn't configure the crate properly for their usage.